controllers/actions.summerwind.net/sync_volumes.go

```diff
@@ -159,27 +159,31 @@ func syncPV(ctx context.Context, c client.Client, log logr.Logger, ns string, pv
 	log.V(2).Info("checking pv phase", "phase", pv.Status.Phase)
 
 	if pv.Status.Phase != corev1.VolumeReleased {
-		// We assume that the pvc is shortly terminated, hence retry forever until it gets removed.
+		// We assume that the PVC is shortly terminated, hence retry forever until it gets removed.
 		retry := 10 * time.Second
-		log.V(1).Info("Retrying sync until pvc gets released", "requeueAfter", retry)
+		log.V(1).Info("Retrying sync until PVC gets released", "requeueAfter", retry)
 		return &ctrl.Result{RequeueAfter: retry}, nil
 	}
 
-	// At this point, the PV is still Released
-
-	pvCopy := pv.DeepCopy()
-	delete(pvCopy.Labels, labelKeyCleanup)
-	pvCopy.Spec.ClaimRef = nil
-	log.V(2).Info("Unsetting PV's claimRef", "pv", pv.Name)
-	if err := c.Update(ctx, pvCopy); err != nil {
-		return nil, err
+	// Check if the PV has ReclaimPolicy "Delete".
+	if pv.Spec.PersistentVolumeReclaimPolicy == corev1.PersistentVolumeReclaimDelete {
+		log.Info("Skipping manipulation for PV with 'Delete' reclaim policy", "pv", pv.Name)
+		// For PVs with ReclaimPolicy "Delete", we don't need to do anything.
+		return nil, nil
 	}
 
-	log.Info("PV should be Available now")
+	// If ReclaimPolicy is not "Delete", we proceed to clean up the ClaimRef.
+	if pv.Status.Phase == corev1.VolumeReleased {
```

Comment:
Sorry for the delayed response!
I believe this "if" is unneeded because we already guard against phases other than Released in L161-L166
Other than that, this looks good to me!
Thanks for fixing it, @midnattsol  ☺️

Reply:
You're right, I just made a commit removing that if.